4. Quel influence a le fait de créer beaucoup de posts sur le comportement des utilisateurs (comparer les gros créateurs avec les petits, par quartile ?).


posts_created_per_week  

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
df = pd.read_csv("./dataset/instagram_users_lifestyle.csv")

df.head()        # voir les premières lignes
df.info()        # types des colonnes
df.describe()    # stats générales

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1547896 entries, 0 to 1547895
Data columns (total 58 columns):
 #   Column                          Non-Null Count    Dtype  
---  ------                          --------------    -----  
 0   user_id                         1547896 non-null  int64  
 1   app_name                        1547896 non-null  object 
 2   age                             1547896 non-null  int64  
 3   gender                          1547896 non-null  object 
 4   country                         1547896 non-null  object 
 5   urban_rural                     1547896 non-null  object 
 6   income_level                    1547896 non-null  object 
 7   employment_status               1547896 non-null  object 
 8   education_level                 1547896 non-null  object 
 9   relationship_status             1547896 non-null  object 
 10  has_children                    1547896 non-null  object 
 11  exercise_hours_per_week         1547896 non-null  float64
 12  

user_id           age  exercise_hours_per_week  \
count  1.547896e+06  1.547896e+06             1.547896e+06   
mean   7.739485e+05  3.898544e+01             7.143480e+00   
std    4.468392e+05  1.529453e+01             3.994556e+00   
min    1.000000e+00  1.300000e+01             0.000000e+00   
25%    3.869748e+05  2.600000e+01             4.000000e+00   
50%    7.739485e+05  3.900000e+01             6.600000e+00   
75%    1.160922e+06  5.200000e+01             9.700000e+00   
max    1.547896e+06  6.500000e+01             2.410000e+01   

       sleep_hours_per_night  perceived_stress_score  self_reported_happiness  \
count           1.547896e+06            1.547896e+06             1.547896e+06   
mean            6.999384e+00            1.998774e+01             5.499804e+00   
std             1.097098e+00            1.183258e+01             2.871232e+00   
min             3.000000e+00            0.000000e+00             1.000000e+00   
25%             6.300000e+00            1.000000e+01             3.000000e+00   
50%             7.000000e+00            2.000000e+01             5.000000e+00   
75%             7.700000e+00            3.000000e+01             8.000000e+00   
max             1.000000e+01            4.000000e+01             1.000000e+01   

       body_mass_index  blood_pressure_systolic  blood_pressure_diastolic  \
count     1.547896e+06             1.547896e+06              1.547896e+06   
mean      2.500084e+01             1.244917e+02              7.950007e+01   
std       3.976188e+00             2.020699e+01              1.154874e+01   
min       1.500000e+01             9.000000e+01              6.000000e+01   
25%       2.230000e+01             1.070000e+02              7.000000e+01   
50%       2.500000e+01             1.240000e+02              7.900000e+01   
75%       2.770000e+01             1.420000e+02              9.000000e+01   
max       4.500000e+01             1.590000e+02              9.900000e+01   

       daily_steps_count  ...  time_on_explore_per_day  \
count       1.547896e+06  ...             1.547896e+06   
mean        8.000105e+03  ...             3.762037e+01   
std         8.939761e+01  ...             2.535788e+01   
min         7.566000e+03  ...             1.000000e+00   
25%         7.940000e+03  ...             1.800000e+01   
50%         8.000000e+03  ...             3.400000e+01   
75%         8.060000e+03  ...             5.300000e+01   
max         8.473000e+03  ...             1.710000e+02   

       time_on_messages_per_day  time_on_reels_per_day  followers_count  \
count              1.547896e+06           1.547896e+06     1.547896e+06   
mean               3.295203e+01           5.647662e+01     2.160543e+03   
std                2.144569e+01           3.534782e+01     3.402016e+03   
min                1.000000e+00           1.000000e+00     1.000000e+01   
25%                1.600000e+01           2.900000e+01     5.430000e+02   
50%                3.100000e+01           5.400000e+01     1.154000e+03   
75%                4.700000e+01           8.000000e+01     2.459000e+03   
max                1.380000e+02           2.220000e+02     2.088110e+05   

       following_count  notification_response_rate  account_creation_year  \
count     1.547896e+06                1.547896e+06           1.547896e+06   
mean      2.604942e+03                5.002404e-01           2.018466e+03   
std       2.751922e+03                2.887827e-01           4.613253e+00   
min       2.000000e+01                0.000000e+00           2.010000e+03   
25%       6.630000e+02                2.500000e-01           2.015000e+03   
50%       1.519000e+03                5.000000e-01           2.019000e+03   
75%       3.451000e+03                7.500000e-01           2.022000e+03   
max       1.000000e+04                1.000000e+00           2.026000e+03   

       average_session_length_minutes  linked_accounts_count  \
count                    1.547896e+06           1.547896e+06   
mean    

On va découper posts_created_per_week en 4 groupes :

Q1 → petits créateurs

Q2 → moyens

Q3 → gros

Q4 → très gros créateurs

In [12]:
import pandas as pd

# Création des quartiles
df['creator_quartile'] = pd.qcut(
    df['posts_created_per_week'],
    3,
    labels=['Q1_Petit', 'Q2_Moyen', 'Q3_Gros']
)

# Vérification
print(df['creator_quartile'].value_counts())

creator_quartile
Q1_Petit    662786
Q3_Gros     476482
Q2_Moyen    408628
Name: count, dtype: int64


In [28]:
import pandas as pd
import numpy as np

# 🔹 Colonnes pertinentes pour l'analyse du comportement
columns_to_use = [
    # Activité / bien-être
    'exercise_hours_per_week', 'sleep_hours_per_night', 'body_mass_index',
    'perceived_stress_score', 'self_reported_happiness',
    
    # Engagement général / social
    'daily_steps_count', 'weekly_work_hours', 'hobbies_count',
    'social_events_per_month', 'books_read_per_year', 'volunteer_hours_per_month',
    'travel_frequency_per_year',
    
    # Activité sur l'application
    'daily_active_minutes_instagram', 'sessions_per_day', 'posts_created_per_week',
    'reels_watched_per_day', 'stories_viewed_per_day', 'likes_given_per_day',
    'comments_written_per_day', 'dms_sent_per_week', 'dms_received_per_week',
    'ads_viewed_per_day', 'ads_clicked_per_day', 'time_on_feed_per_day',
    'time_on_explore_per_day', 'time_on_messages_per_day', 'time_on_reels_per_day',
    'following_count', 'notification_response_rate',
    'average_session_length_minutes', 'user_engagement_score'
]

# 🔹 Nettoyer les données pour posts_created_per_week (quartiles)
df_clean = df.dropna(subset=['posts_created_per_week'])

# 🔹 Création des quartiles Q1 à Q4
df_clean['creator_quartile'] = pd.qcut(
    df_clean['posts_created_per_week'],
    4,
    labels=['Q1_Petit', 'Q2_Bas', 'Q3_Haut', 'Q4_Gros']
)

# 🔹 Sélection des colonnes numériques pertinentes
df_numeric = df_clean[columns_to_use]

# 🔹 Moyennes par quartile
grouped = df_numeric.groupby(df_clean['creator_quartile']).mean()

# 🔹 Calcul écart Q4 - Q1
diff = grouped.loc['Q4_Gros'] - grouped.loc['Q1_Petit']

# 🔹 Création du tableau final
result = pd.DataFrame({
    'Q1_Petit': grouped.loc['Q1_Petit'],
    'Q4_Gros': grouped.loc['Q4_Gros'],
    'Diff_Q4_minus_Q1': diff
})

# 🔹 Arrondir
result = result.round(2)

# 🔹 Trier du plus grand écart au plus petit
result = result.sort_values(by='Diff_Q4_minus_Q1', ascending=False)

print(result)

                                Q1_Petit  Q4_Gros  Diff_Q4_minus_Q1
following_count                  2169.26  3191.76           1022.49
daily_active_minutes_instagram    116.81   260.86            144.05
reels_watched_per_day             119.56   241.42            121.86
likes_given_per_day                83.38   155.45             72.07
time_on_feed_per_day               58.41   130.46             72.05
stories_viewed_per_day             76.06   126.03             49.97
time_on_reels_per_day              35.06    78.25             43.19
time_on_explore_per_day            23.36    52.13             28.77
time_on_messages_per_day           20.46    45.63             25.17
comments_written_per_day           23.51    45.08             21.57
dms_received_per_week              22.02    39.31             17.29
dms_sent_per_week                  21.68    36.08             14.40
perceived_stress_score             13.90    25.56             11.66
ads_viewed_per_day                 14.35    25.8

/tmp/ipykernel_30657/1902291478.py:39: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  grouped = df_numeric.groupby(df_clean['creator_quartile']).mean()


| Métrique                        | Q1_Petit | Q4_Gros | Diff_Q4_minus_Q1 |
|---------------------------------|----------|---------|-----------------|
| following_count                  | 2169.26  | 3191.76 | 1022.49         |
| daily_active_minutes_instagram   | 116.81   | 260.86  | 144.05          |
| reels_watched_per_day            | 119.56   | 241.42  | 121.86          |
| likes_given_per_day              | 83.38    | 155.45  | 72.07           |
| time_on_feed_per_day             | 58.41    | 130.46  | 72.05           |
| stories_viewed_per_day           | 76.06    | 126.03  | 49.97           |

On peut donc voir que ce sont ses données qui change le plus selon le nombre de post qu'ils font

On pourrait prendre ses données pour prédire le nombre de post par jour

On pourrait faire des personnas : gros créateurs et petits créateurs avec ses informations